src/test/java/org/takes/tk/TkProxyTest.java

```diff
@@ -37,19 +39,125 @@
+ *
  *  The tests should verify the different HTTP methods (GET, POST, etc),
  *  as well as the different combinations of request/response headers.
  */
 public final class TkProxyTest {
 
     /**
+     * Test string.
+     */
+    public static final String HELLO_WORLD = "hello, world!";
+
+    /**
+     * Format string.
+     */
+    public static final String FORMAT = "%s:%d";
+
+    /**
+     * Dash constant.
+     */
+    public static final String DASH = "/";
+
+    /**
+     * Response string.
+     */
+    public static final String HEADER = "X-Takes-TkProxy:";
+
+    /**
+     * OK response.
+     */
+    public static final String OK = "HTTP/1.1 200 OK";
+
+    /**
      * TkProxy can work.
      * @throws Exception If some problem inside
      */
     @Test
     public void justWorks() throws Exception {
-        new FtRemote(new TkFixed("hello, world!")).exec(
+        new FtRemote(new TkFixed(HELLO_WORLD)).exec(
+                new FtRemote.Script() {
+                    @Override
+                    public void exec(final URI home) throws IOException {
+                        MatcherAssert.assertThat(
+                                new RsPrint(
+                                        new TkProxy(
+                                                String.format(
+                                                        FORMAT,
+                                                        home.getHost(),
+                                                        home.getPort()
+                                                )
+                                        ).act(new RqFake())
+                                ).print(),
+                                Matchers.containsString(HELLO_WORLD)
+                        );
+                    }
+                }
+        );
+    }
+
+    /**
+     * Verifies TkProxy against HTTP methods (GET, POST, OPTIONS, PUT, DELETE).
+     *
+     * @throws Exception If some problem inside
+     */
+    @Test
+    public void actsOnHttpMethods() throws Exception {
+        for (
+            final String httpMethod : new String[]{
+                Request.GET,
+                Request.POST,
+                Request.OPTIONS,
+                Request.PUT,
+                Request.DELETE,
+            }) {
+            this.acts(httpMethod);
+        }
+    }
+
+    /**
+     * Verifies TkProxy against invalid HTTP methods
+     * (verifies IO Exception is thrown).
+     *
+     * @throws Exception If some problem inside
+     */
+    @Test(expected = IOException.class)
+    public void actsOnInvalidHttpMethods() throws Exception {
+        this.acts("INVALIDHTTPMETHOD");
+    }
+
+    /**
+     * Verifies TkProxy against request headers
+     * (that the original request has been proxied).
+     * (4 random headers)
+     *
+     * @throws Exception If some problem inside
+     */
+    @Test
+    public void actsOnReqHeaders() throws Exception {
```

Comment:
@aschworer please, don't shorten the test name. It's the most useful source of information about test purpose.